### PR TITLE
Address object

### PR DESCRIPTION
**Pull request by wseifstad**
Created address object instead of pandas df

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,16 +1,18 @@
...
 Addresses obtained from https://www.melissadata.com/lookups/zipstreet.asp?InData=11215&c=3&l=U   
 Address list currently only has numbered streets in 11215 zip code.
 
+
 # Installation
 
 Clone the repo onto your machine with the following command:
 
-$ git checkout https://github.com/CloudburstCode/passiv.git
+$ git clone https://github.com/wseife/better-walk.git
```
Comment (wyattshapiro) [inline README.md:16]: lol whoops just copied that one over, good catch

File: src/address_with_days.csv [hunk 2]
```diff
@@ -0,0 +1,4759 @@
+address,trash_day
```
Comment (wyattshapiro) [inline src/address_with_days.csv:1]: aweomse!

File: src/create_objects.py [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+            print("object created")
+            print("-"*30)
+        else:
+            print("object failed")
+            print("-"*30)
+
+    driver.close()
+    
+    #print test to show objects in list:
+    for i in object_list:
```
Comment (wyattshapiro) [inline src/create_objects.py]: delete all these print lines since they are unnecessary

File: src/model.py [hunk 4]
```diff
@@ -0,0 +1,23 @@
+class address_obj(object):
+    """
+    create address object initialized with
+    address, latitude, longitude and trash_days
+    """
+
+    def __init__(self,address,lat,lng,trash_days):
```
Comment (wyattshapiro) [inline src/model.py:7]: perfect!

File: src/model.py [hunk 5]
```diff
@@ -0,0 +1,23 @@
...
+        self.address = address
+        self.lat = lat
+        self.lng = lng
+        self.trash_days = trash_days
+    
+    def get_address(self):
+        return self.address
+
+    def get_lat(self):
```
Comment (wyattshapiro) [inline src/model.py:16]: delete

File: src/model.py [hunk 6]
```diff
@@ -0,0 +1,23 @@
...
+        return self.lat
+    
+    def get_lng(self):
```
Comment (wyattshapiro) [inline src/model.py:19]: delete

File: src/model.py [hunk 7]
```diff
@@ -0,0 +1,23 @@
...
+        return self.lng
+        
+    def get_trash_days(self):
```
Comment (wyattshapiro) [inline src/model.py:22]: delete

File: src/scrape.py [hunk 8]
```diff
@@ -1,35 +1,58 @@
...
-    trash_days_list = []
+    trash_days_dict = {}
     test_flag = False
+    driver = scrape_util.initChromeDriver()
 
     if test_flag:
         trash_days_dict = {}
         test_address = "160 15th Street, Brooklyn, NY, USA"
-        trash_days = scrape_util.getTrashDays(test_address, screenshot=True)
+        trash_days = scrape_util.getTrashDays(test_address, driver, screenshot=False)
```
Comment (wyattshapiro) [inline src/scrape.py]: word!

File: src/scrape.py [hunk 9]
```diff
@@ -1,35 +1,58 @@
...
 
         for index, row in address_list.iterrows():
-            trash_days_dict = {}
             address = str(row["Address"]) + ", Brooklyn, NY, USA"
-            print("checking address %s" % str(index+1))
-            trash_days = scrape_util.getTrashDays(address, screenshot=False)
+            print("checking address %s of %s" % (str(index+1) , len(address_list)))
+            print("program has been running for %s seconds" % str(time.time() - start_time))
+            trash_days = scrape_util.getTrashDays(address, driver, screenshot=False)
             trash_days_dict[address] = trash_days
```
Comment (wyattshapiro) [inline src/scrape.py]: we should be creating objects here, not a dictionary

File: src/constants.py [hunk 10]
```diff
@@ -1 +1 @@
-BASE_DIR = "/Users/wyattshapiro/projects/dog_walk/"
+BASE_DIR = "/Users/williamseife/Documents/GitHub/better-walk/"
```
Comment (wyattshapiro) [inline src/constants.py]: change this to the following:
`import os
BASE_DIR = os.path.dirname(os.path.dirname(os.path.realpath(__file__)))`

This finds the real path of the file (constants.py), gets the directory (src/) and then gets that directory (better-walk/)

File: src/scrape.py [hunk 11]
```diff
@@ -1,35 +1,58 @@
...
-            trash_days_list.append(trash_days_dict)
+        
+        ## clear addresses with NoneType trash_days
+        clean_dict = {k: v for k, v in trash_days_dict.iteritems() if v is not None}
```
Comment (wyattshapiro) [inline src/scrape.py]: why would we have a NoneType trash day? If we do, this should be functionalized and then called inside the getTrashDays

File: src/scrape.py [hunk 12]
```diff
@@ -1,35 +1,58 @@
...
+            print("checking address %s of %s" % (str(index+1) , len(address_list)))
+            print("program has been running for %s seconds" % str(time.time() - start_time))
+            trash_days = scrape_util.getTrashDays(address, driver, screenshot=False)
             trash_days_dict[address] = trash_days
-            trash_days_list.append(trash_days_dict)
+        
+        ## clear addresses with NoneType trash_days
+        clean_dict = {k: v for k, v in trash_days_dict.iteritems() if v is not None} 
+
+        trash_df = pd.DataFrame.from_dict(clean_dict,orient="index") \
```
Comment (wyattshapiro) [inline src/scrape.py]: this should be a list of objects rather than dicts

File: src/util/plot_util.py [hunk 13]
```diff
@@ -0,0 +1,41 @@
...
+        "TUE": "#008000",  #Green
+        "WED": "#0000FF",  #Blue
+        "THU": "#FF00FF",  #Fuschia
+        "FRI": "#FFFF00",  #Yellow
+        "SAT": "#00FFFF",  #aqua
+        "SUN": "#808080",  #grey            
+        }  
+
+def geocode(address):
+    try:
```
Comment (wyattshapiro) [inline src/util/plot_util.py]: word! try and except are great

File: src/create_objects.py [hunk 14]
```diff
@@ -0,0 +1,62 @@
+import pandas as pd
```
Comment (wyattshapiro) [inline src/create_objects.py]: i am confused by this file, why are we plotting? 

we should be scraping in one function, then once we have scraped we can plot!

**Review by wyattshapiro — CHANGES_REQUESTED** [review]
Overall, the object looks great with the init function. Next step: in the scrape.py iterate though each address, scrape trashdays from getTrashDays, create an object for every unique address/day combo, store those list of objects in a master list of objects using .extend(). Finally, we can plot these address objects on a map.

File: src/model.py (hunk 7 above)
Comment (wseifstad) [reply src/model.py:22]: Once we have a list of objects, don't we then need methods to call the object's address, lat, lng and days? I think I'm failing to see the benefit of using objects in this case as opposed to a table. In order to select a specific set of addresses (for example, that fall on a certain trash day), don't we need to call each object's trash_days method?

File: src/scrape.py (hunk 9 above)
Comment (wseifstad) [reply src/scrape.py]: I'm going to delete scrape.py -- it's fully implemented in the create_objects.py script.

File: src/constants.py (hunk 10 above)
Comment (wseifstad) [reply src/constants.py]: done.

File: src/scrape.py (hunk 11 above)
Comment (wseifstad) [reply src/scrape.py]: This was the case for trash days that failed to scrape (i.e. if the address doesn't have any associated trash day, or if the browser fails to load the page). But, again, scrape.py should be ignored.

File: src/scrape.py (hunk 12 above)
Comment (wseifstad) [reply src/scrape.py]: same as above

File: src/create_objects.py (hunk 14 above)
Comment (wseifstad) [reply src/create_objects.py]: bit of a misnomer with plot_util -- I created it for the map_plot, but it also contains the code for geocoding the addresses, hence my calling it for creating the Address objects. We can separate the geocoding into a different file, or rename this script if it's confusing.

Comment (wseifstad) [reply src/create_objects.py]: Here's what I'm going to do:
-rename "create_objects" this script "scrape.py"
functionality will be to scrape and geocode each address (throwing out addresses that fail to scrape or fail to geocode)
-move the geocode function into scrape_util to keep plotting separate.

**Review by wyattshapiro — APPROVED** [review]

**Pull request merged**